iceoryx_posh/include/iceoryx_posh/internal/roudi/service_registry.hpp

```diff
@@ -123,13 +125,17 @@ class ServiceRegistry
   private:
     uint32_t findIndex(const capro::ServiceDescription& serviceDescription) const noexcept;
 
-    void getAll(ServiceDescriptionVector_t& searchResult) const noexcept;
-
     void applyToAll(cxx::function_ref<void(const ServiceDescriptionEntry&)> callable) const noexcept;
 
     cxx::expected<Error> add(const capro::ServiceDescription& serviceDescription,
                              ReferenceCounter_t ServiceDescriptionEntry::*count);
 };
+
+constexpr uint32_t ServiceRegistry::capacity() noexcept
```

Comment:
If you keep the methods, please move the implementation to the `.cpp`

Reply:
Not possible with constexpr methods.

Reply:
Ok, please then create a `service_registry.inl` and move this method to this new file, like I did with `cxx::string`: https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/string.inl#L287-L290

Reply:
I will remove the method, but I think it would be good design to have it. I consider it not worth it if these are our requirements to have it. This does not lead to good designs and reasonable file structure.